addons/website_sale/models/payment_transaction.py

```diff
@@ -10,3 +10,4 @@ def _check_amount_and_confirm_order(self):
         """Override of `sale` to archive guest contacts."""
         confirmed_orders = super()._check_amount_and_confirm_order()
         confirmed_orders.filtered('website_id')._archive_partner_if_no_user()
```

Comment:
Wrongly merged yesterday through #240216 (but had no impact because the file wasn't imported)